### PR TITLE
make cookies injectable and add an extension error report controller

### PR DESCRIPTION
**Pull request by eishay**

### CODE REVIEW

File: shoebox/app/com/keepit/common/healthcheck/Healthcheck.scala [hunk 1]
```diff
@@ -31,6 +31,7 @@ object Healthcheck {
   case object FACEBOOK extends CallType
   case object BOOTSTRAP extends CallType
   case object INTERNAL extends CallType
+  case object EXTERNTION extends CallType
```
Comment (2is10) [inline shoebox/app/com/keepit/common/healthcheck/Healthcheck.scala]: `EXTENSION` (spelling)

File: shoebox/app/com/keepit/controllers/ext/ExtErrorReportController.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    extends BrowserExtensionController(actionAuthenticator) with ShoeboxServiceController {
+
+  def addErrorReport = AuthenticatedJsonToJsonAction { request =>
+    val json = request.body
+    val message = (json \ "message").as[String]
+    val errorReport = healthcheck.addError(
+      HealthcheckError(error = None,
+        callType = Healthcheck.EXTERNTION,
+        errorMessage = Some(s"error on user ${request.userId} extension installation id ${request.kifiInstallationId}" +
+          s" using experimants [${request.experimants.mkString(",")}]" +
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/ext/ExtErrorReportController.scala]: `experiments` (spelling)

File: shoebox/app/com/keepit/controllers/ext/ExtErrorReportController.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          s": $message")
+      )
+    )
+    Ok(JsObject(Seq("error-id" -> JsString(errorReport.id.id))))
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/ext/ExtErrorReportController.scala]: `"errorId"` (our camel-case JSON convention)

File: shoebox/conf/routes [hunk 4]
```diff
@@ -70,6 +70,8 @@ POST    /messages/:parentId         @com.keepit.controllers.ext.ExtCommentContro
 POST    /users/slider/suppress      @com.keepit.controllers.ext.ExtUserController.suppressSliderForSite
 
 POST    /users/events               @com.keepit.controllers.ext.ExtEventController.logUserEvents
+POST    /error/add/authenticated    @com.keepit.controllers.ext.ExtErrorReportController.addErrorReport
+POST    /error/add/unauthenticated  @com.keepit.controllers.ext.ExtErrorReportController.addUnauthenticatedErrorReport
```
Comment (2is10) [inline shoebox/conf/routes]: Andrew added a way to write an action with two handlers... one for the authenticated case and one for the unauthenticated case. Would you mind using it to combine these into one action at `/error` or `/error/report`? (I believe it should be the server's responsibility to determine whether a request is authenticated.) Thanks.

**Comment by 2is10** [comment]
Great! Thanks for getting started on this.

File: shoebox/app/com/keepit/common/healthcheck/Healthcheck.scala (hunk 1 above)
Comment (eishay) [inline shoebox/app/com/keepit/common/healthcheck/Healthcheck.scala]: :-(

**Comment by eishay** [comment]
@2is10 fixed, please review again. we can start sending stuff there as a message and add fields as we go. i'll work on making the healthcheck error be more flexible

**Comment by 2is10** [comment]
Looks great, thanks. :heart: filenames matching class names

**Pull request merged**